### DISCUSSION TITLE
How to get exact mass by specified element in one molecule ?

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#4632] [Q&A] [answered]
The exact mass of the whole molecule can be obtained by method of `ExactMolWt()`
I want to get mass spectrometry data, so I need to get the exact mass of each element in the molecule. what should I do in this case?
By the way, What is the formula for rdkit to calculate the exact numerator of one molecule?
Is there an official data sheet for the exact mass of each element’s atom?
Thank you !

**Comment by dehaenw** [comment, marked as answer]
Hello @ShangChien,
I hope i understood your question correctly.
to get molecular weights corresponding to typical organic mass spectrometry peaks you will need to fragment your molecule. For this you would probably need to create SMARTS patterns corresponding to typical fragmentation patterns you expect in your molecules of interest (e.g. McLafferty rearrangement and such things). Here is an interesting paper, where among other things they use a rules based technique to fragment molecules https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/pdf/10.1002/jms.4278. This is a hard task because for an exhaustive result a fair bit of electron pushing and chemical intuition not easily captured by a model or a rule based flow is needed.

The exact atomic weight per atom can be found in many places, for example you can check the IUPAC numbers here https://www.ciaaw.org/atomic-weights.htm, The linked ciaaw site also has isotope abundances and a CSV with atomic mass per isotope https://www.ciaaw.org/atomic-masses.htm.

**Reply by ShangChien** [reply]
thank you for you useful answer!

**Comment by bp-kelley** [comment]
You may also find rdkit’s periodic table useful

https://www.rdkit.org/docs/source/rdkit.Chem.PeriodicTable.html

----
Brian Kelley

**Reply by ShangChien** [reply]
nice document!
